packages/side-nav/src/SideNav.tsx

```diff
@@ -35,11 +161,116 @@ interface SideNavProps {
  * @param props.className Class name that will be applied to the root-level element.
  * @param props.children Content that will be rendered inside the root-level element.
  */
-function SideNav({ className, children, ...rest }: SideNavProps) {
+function SideNav({ className, children, id: idProp, ...rest }: SideNavProps) {
+  const { Provider: ContextProvider } = SideNavContext;
+  const [collapsed, setCollapsed] = useState(false);
+  const [hover, setHover] = useState(false);
+  const [focus, setFocus] = useState(false);
+  const { usingKeyboard } = useUsingKeyboardContext();
+  const navId = useMemo(() => idProp ?? navIdAllocator.generate(), [idProp]);
+  const [
+    portalContainer,
+    setPortalContainer,
+  ] = useState<HTMLUListElement | null>(null);
+
+  // We visually expand the navigation when a user focuses on an element within the navigation
+  // while navigating via keyboard.
+  const focusExpand = usingKeyboard && focus;
+
+  // Nav element should have a programmatically-determinable label
+  validateAriaLabelProps(rest, 'SideNav');
```

Comment:
I feel like this is a relatively stable component, it might be possible that we just provide teh default label of "Side Navigation." In storybook, I believe the aria-label supplied was "Realm" or something to that effect, and I actually think the function of the component, rather than its context, might be more descriptive here

Reply:
I'll see if I can dig it up, but the guidelines I was following specifically said *not* to include the words "navigation" and to just say at a high level what you're navigating. E.g. "Realm Project" instead of "Realm Project Side Nav".

Reply:
Interesting